pkg/apis/options/alpha_options.go

```diff
@@ -42,7 +47,9 @@ type AlphaOptions struct {
 	MetricsServer Server `json:"metricsServer,omitempty"`
 
 	// Providers is used to configure multiple providers.
-	Providers Providers `json:"providers,omitempty"`
+	Providers provideropts.Providers `json:"providers,omitempty"`
```

Comment:
Why has this been moved into a new package?

Reply:
otherwise get import cycles,

```
package github.com/oauth2-proxy/oauth2-proxy/v7
        imports github.com/oauth2-proxy/oauth2-proxy/v7/pkg/apis/options
        imports github.com/oauth2-proxy/oauth2-proxy/v7/tenant
        imports github.com/oauth2-proxy/oauth2-proxy/v7/tenant/configloader
        imports github.com/oauth2-proxy/oauth2-proxy/v7/pkg/apis/options: import cycle not allowed
```

basically ConfigLoader (a TenantLoader implementation) config depends on the provider config, which is part of `pkg/apis/options` package, and the options.Options depend on the TenantLoader config which depends upon the ConfigLoader config.

This is resolved by moving the provider options to a new package.

Reply:
> Why has this been moved into a new package?

resolved. moved to the original location.

Reply:
> Why has this been moved into a new package?

resolved. moved to the original location.